tools/harness-simulation/posix/sniffer_sim/pcap_codec.py

```diff
@@ -89,11 +88,7 @@ def append(self, frame):
             return
 
         timestamp = self._get_timestamp()
-        pkt = self.encode_frame(frame, *timestamp)
-        self._pcap_file.write(pkt)
-        self._pcap_file.flush()
-
-    def close(self):
-        """ Close the pcap file. """
+        self._pcap_content += self._encode_frame(frame, *timestamp)
```

Comment:
It saves pcap in the memory, and the pcap might be large.

Should we be concerned of the memory footprint?

Reply:
BTW, the string concatenation gets less and less efficient as the pcap grows larger.

Reply:
> It saves pcap in the memory, and the pcap might be large.
> 
> Should we be concerned of the memory footprint?

Referenced to the previous test results, the largest PCAP capture file generated is less than 300 KiB. The memory usage should not be a concern if we run the sniffer simulation on a Raspberry Pi.

Reply:
> BTW, the string concatenation gets less and less efficient as the pcap grows larger.

I will append each frame into a list, and then join them to get the output for better performance in this PR.
In next PR, each frame will be forwarded to `tshark` via a named pipe.

Reply:
> In next PR, each frame will be forwarded to tshark via a named pipe.

That should also fix the memory footprint issue?

Reply:
> That should also fix the memory footprint issue?

I think the answer is yes. There will be no memory footprint issue on my side, and `tshark` should handle the streamed sent packets properly.